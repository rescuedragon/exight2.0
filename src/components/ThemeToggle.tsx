import { Moon, Sun } from 'lucide-react';
import { Button } from '@/components/ui/button';
import { useTheme } from 'next-themes';

export const ThemeToggle = () => {
  const { theme, setTheme } = useTheme();

  return (
    <Button
      variant="outline"
      size="icon"
      onClick={() => setTheme(theme === 'light' ? 'dark' : 'light')}
      className="h-12 w-12 rounded-full bg-gradient-card border-border/40 shadow-card hover:shadow-elevated transition-all duration-200 hover:scale-102"
    >
      <Sun className="h-5 w-5 rotate-0 scale-100 transition-all dark:-rotate-90 dark:scale-0 text-orange-accent" />
      <Moon className="absolute h-5 w-5 rotate-90 scale-0 transition-all dark:rotate-0 dark:scale-100 text-blue-accent" />
      <span className="sr-only">Toggle theme</span>
    </Button>
  );
};
