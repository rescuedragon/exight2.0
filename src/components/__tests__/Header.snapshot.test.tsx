import { describe, it, expect } from 'vitest';
import { render } from '@testing-library/react';
import { ThemeToggle } from '../ThemeToggle';

function Header() {
  return (
    <div className="flex items-center justify-between">
      <h1 className="text-xl font-bold">Exight</h1>
      <ThemeToggle />
    </div>
  );
}

describe('Header snapshot', () => {
  it('renders stable markup', () => {
    const { container } = render(<Header />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
