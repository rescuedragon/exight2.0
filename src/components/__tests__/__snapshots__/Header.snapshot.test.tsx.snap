// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header snapshot > renders stable markup 1`] = `
<div
  class="flex items-center justify-between"
>
  <h1
    class="text-xl font-bold"
  >
    Exight
  </h1>
  <button
    class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 border hover:bg-accent hover:text-accent-foreground h-12 w-12 rounded-full bg-gradient-card border-border/40 shadow-card hover:shadow-elevated transition-all duration-200 hover:scale-102"
  >
    <svg
      class="lucide lucide-sun h-5 w-5 rotate-0 scale-100 transition-all dark:-rotate-90 dark:scale-0 text-orange-accent"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="12"
        cy="12"
        r="4"
      />
      <path
        d="M12 2v2"
      />
      <path
        d="M12 20v2"
      />
      <path
        d="m4.93 4.93 1.41 1.41"
      />
      <path
        d="m17.66 17.66 1.41 1.41"
      />
      <path
        d="M2 12h2"
      />
      <path
        d="M20 12h2"
      />
      <path
        d="m6.34 17.66-1.41 1.41"
      />
      <path
        d="m19.07 4.93-1.41 1.41"
      />
    </svg>
    <svg
      class="lucide lucide-moon absolute h-5 w-5 rotate-90 scale-0 transition-all dark:rotate-0 dark:scale-100 text-blue-accent"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 3a6 6 0 0 0 9 9 9 9 0 1 1-9-9Z"
      />
    </svg>
    <span
      class="sr-only"
    >
      Toggle theme
    </span>
  </button>
</div>
`;
